firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteSchema.java

```diff
@@ -365,45 +364,39 @@ private void createV1RemoteDocumentCache() {
    */
   private void createFieldIndex() {
     ifTablesDontExist(
-        new String[] {"index_configuration", "index_entries"},
+        new String[] {"index_configuration", "index_state", "collection_state", "index_entries"},
         () -> {
-          // TODO(indexing): Do we need to store a different update time per user? We need to ensure
-          // that we index mutations entries for all users.
+          // Global configuration for all existing indices
           db.execSQL(
               "CREATE TABLE index_configuration ("
                   + "index_id INTEGER, "
                   + "collection_group TEXT, "
                   + "index_proto BLOB, " // V1 Admin index proto
-                  + "active INTEGER, " // whether index is active
-                  + "update_time_seconds INTEGER, " // time of last document update added to index
-                  + "update_time_nanos INTEGER, "
                   + "PRIMARY KEY (index_id))");
 
-          // The index entries table only has a single primary index. `array_value` should be set
-          // for all queries.
+          // Per index per user state to track the backfill state for each index
+          db.execSQL(
+              "CREATE TABLE index_state ("
+                  + "uid TEXT, "
+                  + "index_id INTEGER, " // Name of the collection group.
+                  + "sequence_number INTEGER, " // Specifies the order of updates
+                  + "read_time_seconds INTEGER, " // Time of last index backfill update
```

Comment:
nit: be more specific with comment: last processed RDC read time of last backfill update 

currently, there's some confusion with which time is being stored